### ISSUE TITLE
Basic variable layout renderer doesn't seem to be supported

### ISSUE THREAD

**Issue by jlikens** [justeat/NLog.StructuredLogging.Json#76]
I'm trying to output basic variables as in:

> 
    <variable name="foo" value="X.Y.Z" />
    <variable name="bar" value="A-B-C" />
    <extensions>
      <add assembly="NLog.Web" />
      <add assembly="NLog.StructuredLogging.Json" />
    </extensions>
    <targets>
      <target type="BufferingWrapper" name="textBuffer" bufferSize="100" flushTimeout="5000">
        <target name="jsonFile" type="File" fileName="${basedir}\log.json" archiveAboveSize="104857600" maxArchiveFiles="10" archiveNumbering="Sequence">
          <layout type="JsonWithProperties">
            <property name="foo" layout="${var:foo}" />
            <property name="bar" layout="${var:bar}" />
            <property name="machineName" layout="${machinename}" />
          </layout>
        </target>
      </target>
    <rules>
      <logger name="*" minlevel="Trace" writeTo="textBuffer" />
    </rules>

I threw the machineName one in there to check my sanity.  The log output has all of the StructureLogging properties plus machineName, but foo and bar are nowhere to be found (not even an empty property).  I also tried tinkering with the variable layout renderer by specifying "${var:name=foo}" to no avail.

Is this expected behavior?  If so, is there a known workaround?

**Comment by AnthonySteele** [comment]
This might be a bug or a missing feature, as these things should be all wired in.
I'll try and get a  repro in a test case some time soon.

**Comment by jlikens** [comment]
Thanks for the quick response.  I need this fixed ASAP, so I'll probably do a fork later this afternoon and attempt to fix.  If I find anything, I'll submit a pull request and report back here.

**Comment by AnthonySteele** [comment]
That would be great if you do find a fix.

**Comment by AnthonySteele** [comment]
So here's a test for it, which I think is a valid repro of what you describe: https://github.com/AnthonySteele/NLog.StructuredLogging.Json/commit/06f93af4167516a58514aa3d80b28299c416a0bf

It goes wrong in [JsonWithPropertiesLayout.cs](https://github.com/justeat/NLog.StructuredLogging.Json/blob/master/src/NLog.StructuredLogging.Json/JsonWithPropertiesLayout.cs) line 60 - for whatever reason the var value is not picked up, it gets an empty string value instead and this is discarded on line 69.

**Comment by jlikens** [comment]
It appears it's because the Layout doesn't have its LoggingConfiguration property populated, so there are no variables in scope at render time.  I'm still digging around to see why StructuredLoggingProperty objects behave this way.

**Comment by snakefoot** [comment]
@AnthonySteele  & @jlikens Think you are missing the NLogConfigurationItemAttribute:

```
[NLog.Config.NLogConfigurationItem]
public class StructuredLoggingProperty
```

This signals to NLog that it is safe to perform reflection, and scan for properties that should be initialized.

**Comment by AnthonySteele** [comment]
... is the correct answer. here it is in a PR with a test around it https://github.com/justeat/NLog.StructuredLogging.Json/pull/91

**Comment by AnthonySteele** [comment]
fixed by #91
